AppController/djinn.rb

```diff
@@ -1901,13 +1853,11 @@ def job_start(secret)
       # Check the running, terminated, pending AppServers.
       check_running_appservers
 
-      # Detect applications that have been undeployed and terminate all
-      # running AppServers.
-      check_stopped_apps
-
-      # Load balancers and shadow need to check/update nginx/haproxy.
+      # Load balancers and shadow need to check/update applications that have
+      # been undeployed and nginx/haproxy.
       if my_node.is_load_balancer?
         APPS_LOCK.synchronize {
+          check_stopped_apps
```

Comment:
I think the shadow also needs to run this.

Reply:
The shadow is a load_balancer

Reply:
It might be right now by coincidence, but it's a separate role conceptually. There shouldn't be anything that prevents us from assigning instance locations from a machine that's not a load balancer.